### DISCUSSION TITLE
Convert and 23.0.0

### DISCUSSION THREAD

**Discussion by rmoergeli** [JoshClose/CsvHelper#1709] [Q&A] [answered]
I had to update my code recently very often, because every update something else was not working anymore. Hard to find new examples somewhere.

For: 
`            Map(m => m.ComputerName).Convert(row => row.GetField("Computer Name").ToUpper());`

I expect this is now correct?
`            Map(m => m.ComputerName).Convert(field => field.Value.ComputerName.ToUpper());`

But here I'm lost (CustomDateTimeConverter returns a DateTime):
```
          Map(m => m.OperatingSystemInstallDateOrigin).Name("Operating System Install Date").Convert(row =>
          {
              string _operatingSystemInstallDate = row.GetField<string>("Operating System Install Date");
  
              return CustomDateTimeConverter(_operatingSystemInstallDate);
          });

**Comment by JoshClose** [comment, marked as answer]
It seems Visual Studio intellisense has a hard time figuring out which `Convert` object to use and it just merges them into a single one. You should be able to see `ConvertFromStringArgs.Row` when it pops up though. `Value` is used when writing.

Maybe I should change from `Convert` to two methods:
- ConvertFromString
- ConvertToString

This would probably be less confusing as intellisense would know what to do and supply only the correct parameters.

I'm planning on having a migration page in the documentation in the future so people will know exactly what to do to get their code to work when there are breaking changes. I'm in the middle of redoing the documentation though because the current docs library is a major pain and I found something a lot easier to use, and a lot more flexible.

**Reply by JoshClose** [reply]
The change to your code should just be:
```cs
.Convert(args => args.Row.GetField("Computer Name").ToUpper());
```

**Reply by rmoergeli** [reply]
Your example did work.

And the second one works too:
```
            Map(m => m.OperatingSystemInstallDateOrigin).Name("Operating System Install Date").Convert(args =>
                CustomDateTimeConverter(args.Row.GetField("Operating System Install Date")));
```
